### PR TITLE
ensure new endpoint is registered before testing it

### PR DESCRIPTION
**Pull request by bparees**

### CODE REVIEW

**Comment by bparees** [comment]
/test extended_image_ecosystem

@openshift/devex ptal

File: test/extended/image_ecosystem/s2i_perl.go [hunk 1]
```diff
@@ -81,6 +82,27 @@ var _ = g.Describe("[image_ecosystem][perl][Slow] hot deploy for openshift perl
...
+			o.Expect(err).NotTo(o.HaveOccurred())
+
+			// Ran into an issue where we'd try to hit the endpoint before it was updated, resulting in
+			// request timeouts against the previous pod's ip.  So make sure the endpoint ip has changed before
+			// hitting it.
+			err = wait.Poll(1*time.Second, 1*time.Minute, func() (bool, error) {
+				newEndpoint, err := oc.KubeFramework().ClientSet.Core().Endpoints(oc.Namespace()).Get(dcName, metav1.GetOptions{})
+				o.Expect(err).NotTo(o.HaveOccurred())
+				if err != nil {
+					return false, err
```
Comment (jim-minter) [inline test/extended/image_ecosystem/s2i_perl.go]: cannot be hit because of prior `o.Expect(err).NotTo(o.HaveOccurred())`

**Comment by jim-minter** [comment]
@bparees one minor issue (which occurs in multiple files, maybe we should see if this can be refactored one day) then lgtm

File: test/extended/image_ecosystem/s2i_perl.go (hunk 1 above)
Comment (bparees) [reply test/extended/image_ecosystem/s2i_perl.go]: thanks, removed the expects (err is handled outside the wait block)

**Comment by bparees** [comment]
> @bparees one minor issue (which occurs in multiple files, maybe we should see if this can be refactored one day) then lgtm

yeah, I gladly admit there is room for further improvement.

**Comment by bparees** [comment]
this is going to fail until the junit issue is fixed, and the junit PR is going to fail until this is fixed, but this has passed the relevant test, so manually merging.

**Pull request merged**